### ISSUE TITLE
Methods for Github's search API

### ISSUE THREAD

**Issue by essenciary** [JuliaWeb/GitHub.jl#53]
As far as I've seen there are no methods for working with Github's search API. It would be useful to be able to perform random searches and get back a list of repos. 

https://developer.github.com/v3/search/

Cheers!

**Comment by juliohm** [comment]
Indeed, this is all I need right now. There is no way to perform queries about users, repos and orgs with GitHub.jl?

**Comment by jrevels** [comment]
You can make API calls to grab user/repo/org information, but the search API isn't implemented. PRs welcome!

**Comment by cmcaine** [comment]
Are PRs still welcome for this? Is there anything I should know before writing a PR (e.g. should I just copy the structure of the existing code or is there a better way to do it now?)
